onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -299,7 +331,12 @@ class AttentionCPUBase : public AttentionBase {
               // Concatenate past_V and V: (BxNx)PxH_v, (BxNx)LxH_v -> (BxNx)TxH_v
               v = ConcatStateChunk(past, v, present, past_chunk_length, present_chunk_length, i);
             } else if (nullptr != present_value) {
-              v = ConcatStateChunk(past_value, v, present_value, past_chunk_length, present_chunk_length, i);
+              if (use_dmmha) {
```

Comment:
Maybe change if (use_dmmha) to if (past_present_share_buffer). That could be more generic